src/store/coinStore/CoinStore.ts

```diff
@@ -0,0 +1,66 @@
+import { CoinModel } from "@store/models/coin";
+import { Meta } from "@utils/meta";
+import { ILocalStore } from "@utils/useLocalStore";
+import {
+  action,
+  computed,
+  makeAutoObservable,
+  observable,
+  runInAction,
+} from "mobx";
+
+import { requestChart } from "./requestChart";
+import { requestCoin } from "./requestCoin";
+
+export default class CoinStore implements ILocalStore {
+  _id: string | undefined;
+  _coin: CoinModel | undefined;
+  _chart: number[][] | undefined = [];
+  meta: Meta = Meta.initial;
+  constructor(id: string | undefined) {
+    this._id = id;
+    makeAutoObservable(this, {
+      _coin: observable,
+      _chart: observable,
+      fetch: action.bound,
+      chart: computed,
+    });
+  }
+
+  async fetch() {
+    if (this.meta === Meta.loading) return;
+    this.meta = Meta.loading;
+    if (this._id) {
+      const { isError, data } = await requestCoin(this._id);
+      if (isError) {
+        return;
+      }
```

Comment:
возможно из-за этого возникает баг с бесконечным лоадером о котором я писал выше
```suggestion
      if (isError) {
        this.meta = Meta.error;
        return;
      }
```